extensions/resource-deployment/src/ui/resourceTypeWizard.ts

```diff
@@ -0,0 +1,157 @@
+	}
+
+	public async open(): Promise<void> {
+		this.model.initialize();
+		this.wizardObject.generateScriptButton.hidden = true; // by default generateScriptButton stays hidden.
+		this.wizardObject.customButtons = this.customButtons;
+		this.toDispose.push(this.wizardObject.onPageChanged(async (e) => {
+			let previousPage = this.pages[e.lastPage];
+			let newPage = this.pages[e.newPage];
+			//if we are changing to the first page from no page before, essentially when we load the wizard for the first time, e.lastPage is -1 and previousPage is undefined.
+			await previousPage?.onLeave(new WizardPageInfo(e.lastPage, this.pages.length));
+			if (this.useGenerateScriptButton) {
+				if (newPage === this.pages.slice(-1)[0]) {
+					// if newPage is the last page
+					this.wizardObject.generateScriptButton.hidden = false; //un-hide generateScriptButton on last page
+				} else {
+					// if newPage is not the last page
+					this.wizardObject.generateScriptButton.hidden = true; //hide generateScriptButton if it is not the last page
+				}
+			}
+			await newPage.onEnter(new WizardPageInfo(e.newPage, this.pages.length));
+		}));
+
+		this.toDispose.push(this.wizardObject.doneButton.onClick(async () => {
+			await this.model.onOk();
+			this.dispose();
+		}));
+		this.toDispose.push(this.wizardObject.generateScriptButton.onClick(async () => {
+			await this.model.onGenerateScript();
+			this.dispose();
+			this.wizardObject.close(); // close the wizard. This is already hooked up into doneButton, so it is not needed for that button above.
+		}));
+		this.toDispose.push(this.wizardObject.cancelButton.onClick(() => {
+			this.model.onCancel();
+			this.dispose();
+		}));
+
+		await this.wizardObject.open();
+	}
+
+	public addButton(button: azdata.window.Button) {
+		this.customButtons.push(button);
+	}
+
+	public setPages(pages: ResourceTypePage[]) {
+		this.wizardObject!.pages = pages.map(p => p.pageObject);
+		this.pages = pages;
+		this.pages.forEach((page) => {
+			page.pageObject.onValidityChanged((isValid: boolean) => {
+				// generateScriptButton is enabled only when the page is valid.
+				this.wizardObject.generateScriptButton.enabled = isValid;
+			});
+			page.initialize();
+		});
+	}
+
+	protected dispose() {
+		let errorOccurred = false;
+		this.toDispose.forEach((disposable: vscode.Disposable) => {
+			try {
+				disposable.dispose();
+			}
+			catch (error) {
+				errorOccurred = true;
+				console.error(error);
+			}
+		});
+
+		if (errorOccurred) {
+			vscode.window.showErrorMessage(localize('resourceDeployment.DisposableError', "Error occurred while closing the wizard: {0}, open 'Debugger Console' for more information."), this.wizardObject.title);
+		}
+	}
+
+	public registerDisposable(disposable: vscode.Disposable): void {
+		this.toDispose.push(disposable);
+	}
+
+}
+
+
+
+export class ResourceTypePage extends WizardPageBase<ResourceTypeWizard>{
+	public initialize(): void {
+		throw new Error('Method not implemented.');
```

Comment:
This class should probably be abstract instead and then not have a definition for initialize - having this throw isn't really going to ever be a valid scenario I think